datafusion/core/src/physical_plan/file_format/mod.rs

```diff
@@ -698,6 +699,7 @@ mod tests {
             projection,
             statistics,
             table_partition_cols,
+            config_options: ConfigOptions::new().into_shareable(),
```

Comment:
Not sure I understand why we need the `into_shareable` here. Seems like this should just be an owned `ConfigOptions`

Reply:
I agree it is non obvious -- `ConfigOptions` was changed to be shareable  by @waitingkuo  in https://github.com/apache/arrow-datafusion/pull/3455

Basically the usecase there was so that the configuration was owned by `SessionContext` but other parts could read it if necessary -- specifically,  `information_schema.df_settings` / `SHOW ALL` initially. This PR extends the concept so that the settings can be read by the parquet reader

What would you think about moving the mutability handling into `ConfigOption` so this looks like

```rust
            config_options: ConfigOptions::new(),
```

That would hide the details of shared ownership more nicely I think

Reply:
Yeah, that makes sense. If `ConfigOptions` is meant to be shareable then it can just hold a `Arc<RwLock<HashMap<_>>`

Reply:
Will do as a follow on PR

Reply:
Filed https://github.com/apache/arrow-datafusion/issues/3886

Reply:
PR to clean up: https://github.com/apache/arrow-datafusion/pull/3909